### DISCUSSION TITLE
Best practices for adding a port that is utilizing vcpkg manifest mode?

### DISCUSSION THREAD

**Discussion by Znurre** [microsoft/vcpkg#33673] [Q&A] [answered]
Hi,

I have a project, in its own repository, which is using vcpkg manifest mode as per recommendations. Let's refer to it as X. This works all fine.

Now I have another project, let's call it Y, which is also using vcpkg in manifest mode, and from which I want to use project X as a dependency.

I am confused, and cannot find any good info, about what is the best practice for adding project X as a dependency for project Y. Project X already has a `vcpkg.json`, since it's using manifest mode, and that JSON-file could literally be used as a `vcpkg.json` for the port, since it has all info (name, version, dependencies, etc).

Is the intention that port maintainers simply copy the `vcpkg.json` from project X, and use that together with a portfile? Making sure to always keep it up to date with what is in the repo of project X? Or is there some other way one can actually use the `vcpkg.json` file from the repo of project X? I mean, technically I guess I could use git submodules to just add the other repo as a ports sub directory, but that seems to partially defeat the purpose of vcpkg.

Any pointers in the right direction would be appreciated :pray:

**Comment by Osyotr** [comment, marked as answer]
> Is the intention that port maintainers simply copy the `vcpkg.json` from project X

Yes, except that in general it would not be a simple copy because project's manifest may not contain name/version info that is required to build a port.
